woocommerce/woocommerce-ios
# [Mobile Payments] Auto-refund TTP trial payments

## Description


In the original implementation of the Try out Tap to Pay flow, we intended to automatically refund the payment which the merchant makes to themselves, to save them a few steps. This was cut for time on iOS, and completed on Android.

This PR adds the auto-refund feature in the same way as we do on Android

## Testing instructions


1. Launch the app
2. Switch to a Woo Payments US/UK store
3. Navigate to `Menu > Payments`
4. Tap `Set up` or `Try out Tap to Pay on iPhone`
5. Follow the flow through to try a payment
6. Tap `Save receipt and continue`
7. Observe that at the end, you're returned to the payments menu and a Notice is shown to show that the refund was successful.

### Failure

Set up a breakpoint in Proxyman or Charles for POST requests to https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/*/rest-api/, but leave it disabled

1. Repeat the above test, up to step 5
2. Before tapping `Save receipt and continue`, enable your breakpoint
3. Abort (with an error) the request when it comes into your proxy tool
4. Observe that the order details are shown, to allow you to refund the payment manually.

## Screenshots


### Success

https://github.com/woocommerce/woocommerce-ios/assets/2472348/c971726f-2419-4485-aa7d-f0aa8a50632f



### Failure

https://github.com/woocommerce/woocommerce-ios/assets/2472348/0bba1cee-d272-4523-b86b-f42eb0a3b870


---
